playwright/pages/shared/TypeBuilderPage.ts

```diff
@@ -228,10 +228,19 @@ export class TypeBuilderPage extends BuilderPage {
     name: string;
     expectedId: string;
     groupFieldId?: string;
+    grandparentGroupFieldId?: string;
   }) {
-    const { type, name, expectedId, groupFieldId } = args;
+    const { type, name, expectedId, groupFieldId, grandparentGroupFieldId } =
+      args;
 
-    if (groupFieldId) {
+    if (groupFieldId && grandparentGroupFieldId) {
+      await this.getListItem(groupFieldId, grandparentGroupFieldId)
+        .getByRole("button", {
+          name: "Add Field",
+          exact: true,
+        })
+        .click();
```

Comment:
💡 #idea: I think we can simplify this `if...else` block to:

```ts
if (groupFieldId) {
  await this.getListItem(groupFieldId, grandparentGroupFieldId)
    .getByRole("button", {
      name: "Add Field",
      exact: true,
    })
    .click();
} else {
  await this.staticZoneAddFieldButton.click();
}
```

Reply:
Good call.